docs/faq.rst

```diff
@@ -23,5 +23,20 @@ OpenSSL. You're seeing this error because your copy of pip is too old to find
 our wheel files. Upgrade your copy of pip with ``pip install -U pip`` and then
 try install ``cryptography`` again.
 
+Starting ``cryptography`` using ``mod_wsgi`` produces an ``InternalError`` during a call in ``_register_osrandom_engine``
+-------------------------------------------------------------------------------------------------------------------------
+
+This happens because ``mod_wsgi`` uses sub-interpreters, which can cause a
+problem during initialization of the OpenSSL backend. To resolve this set the
+`WSGIApplicationGroup`_ to ``%{GLOBAL}`` in the ``mod_wsgi`` configuration.
+
+``cryptography`` raised an ``InternalError`` and I'm not sure what to do?
+-------------------------------------------------------------------------
+
+Frequently ``InternalError`` is raised when there are errors on the OpenSSL
+error stack that were placed there by other libraries that are also using
+OpenSSL. Try removing the other libraries and see if the problem persists.
```

Comment:
"If you have no other libraries using OpenSSL in your process, or they do
not appear to be at fault, it's possible that this is a bug in
``cryptography``, please file an issue with instructions on how to
reproduce it."

On Sat, Mar 12, 2016 at 3:30 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/faq.rst
> <https://github.com/pyca/cryptography/pull/2817#discussion_r55927221>:
>
> > @@ -23,5 +23,20 @@ OpenSSL. You're seeing this error because your copy of pip is too old to find
> >  our wheel files. Upgrade your copy of pip with ``pip install -U pip`` and then
> >  try install ``cryptography`` again.
> >
> > +Starting ``cryptography`` using ``mod_wsgi`` produces an ``InternalError`` during a call in ``_register_osrandom_engine``
> > +-------------------------------------------------------------------------------------------------------------------------
> > +
> > +This happens because ``mod_wsgi`` uses sub-interpreters, which can cause a
> > +problem during initialization of the OpenSSL backend. To resolve this set the
> > +`WSGIApplicationGroup`_ to ``%{GLOBAL}`` in the ``mod_wsgi`` configuration.
> > +
> > +``cryptography`` raised an ``InternalError`` and I'm not sure what to do?
> > +-------------------------------------------------------------------------
> > +
> > +Frequently ``InternalError`` is raised when there are errors on the OpenSSL
> > +error stack that were placed there by other libraries that are also using
> > +OpenSSL. Try removing the other libraries and see if the problem persists.
>
> Yeah probably. How about "If the error continues to occur please file an
> issue with pyca/cryptography providing whatever information we'll need to
> reproduce it."
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2817/files#r55927221>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084